Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs

```diff
@@ -386,19 +386,93 @@ private void SetMatchingElementDeltaDetails(ElementInfo existingElement, Element
+                if (deletedElement.ElementStatus == ElementInfo.eElementStatus.Failed)
+                {
+                    bool matchingElementFound = false;
+
+                    DeltaElementInfo toRemoveAddedFoundItem = null;
+                    foreach (var newElement in addedElements)
+                    {
+                        var newElementLocdiff = newElement.Locators.ToDictionary(x => x.LocateBy, x => x.LocateValue);
+                        var oldElementLocdiff = deletedElement.Locators.ToDictionary(x => x.LocateBy, x => x.LocateValue);
+
+                        if (newElementLocdiff.Except(oldElementLocdiff).Count() == 0)
+                        {
+                            //check property and update
+                            var newElementPropeties = newElement.Properties.ToDictionary(x => x.Name, x => x.Value);
+                            var oldElementPropeties = deletedElement.Properties.ToDictionary(x => x.Name, x => x.Value);
+                            var diffProperties = newElementPropeties.Except(oldElementPropeties);
+
+                            // check if Parent iframe changed
+                            var parentIFrame = diffProperties.Where(x => x.Key.Contains("Parent IFrame")).FirstOrDefault();
+                            
+                            var deltaControlProp = CovertToDeltaControlProperty(deletedElement.Properties);
+                            if (!string.IsNullOrEmpty(parentIFrame.Value))
+                            {
+                                foreach (var changedProprty in diffProperties.ToList())
+                                {
+                                    foreach (var existingProperty in deltaControlProp.Where(x => x.ElementProperty.Name == changedProprty.Key))
+                                    {
+                                        existingProperty.ElementProperty.Value = changedProprty.Value;
+                                        existingProperty.DeltaStatus = eDeltaStatus.Avoided;
+                                        existingProperty.DeltaExtraDetails = "Property changed";
+                                    }
+                                }
+                            }
+                            matchingElementFound = true;
+
+                            var mathchedItemIndex = DeltaViewElements.IndexOf(DeltaViewElements.Where(x => x.ElementInfo.Guid.Equals(newElement.ElementInfo.Guid)).FirstOrDefault());
+                            //update path of element
+                            deletedElement.Path = newElement.ElementInfo.Path;
+
+                            var item = ConvertElementToDelta(deletedElement, eDeltaStatus.Changed, deletedElement.ElementGroup, true, "Property Changed");
+                            item.Properties = deltaControlProp;
+                            if (mathchedItemIndex != -1)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4592337)